torch/utils/_pytree.py

```diff
@@ -839,6 +839,19 @@ def __repr__(self, indent: int = 0) -> str:
         repr_suffix: str = f"{children_specs_str}])"
         return repr_prefix + repr_suffix
 
+    def __eq__(self, other: PyTree) -> bool:
+        if self is other:
+            return True
+        elif other.__class__ is self.__class__:
+            if str(self.type) != str(other.type):
```

Comment:
I don't think we should heavily rely on the implementation details of `TreeSpec` object. Instead, the better solution is not to allow serializing local classes. For example, the stdlib `pickle` will raise errors for serializing local classes and local functions.